microsoft/testsuites/core/cpu.py

```diff
@@ -0,0 +1,71 @@
+from lisa import testsuite
+from lisa.base_tools.cat import Cat
+from lisa.base_tools.uname import KernelVersion, Uname
+from lisa.node import Node
+from lisa.tools.lscpu import Lscpu
+from lisa.util import SkippedException
+
+NUMA_OFF = "numa=off"
+
+
+@testsuite.TestSuiteMetadata(
+    area="cpu",
+    category="functional",
+    description="""
+    This test suite is used to run CPU related tests.
+    """,
+)
+class CPU(testsuite.TestSuite):
+    @testsuite.TestCaseMetadata(
+        description="""
+        This test case will check that L3 cache is correctly mapped
+        to NUMA node.
+        Steps:
+        1. Check if NUMA is disabled in commandline. If disabled,
+        and kernel version is <= 2.6.37, test is skipped as kernel
+        has no support for NUMA.
+        2. Get the mappings using command :
+        `lscpu --extended=cpu,node,socket,cache`
+        3. Each line in the mapping corresponds to one CPU core. The L3
+        cache of each core must be mapped to the NUMA node that core
+        belongs to instead of the core itself.
+
+        Example :
+        Correct mapping:
+        CPU NODE SOCKET L1d L1i L2 L3
+        8   0    0      8   8   8  0
+        9   1    1      9   9   9  1
+
+        Incorrect mapping:
+        CPU NODE SOCKET L1d L1i L2 L3
+        8   0    0      8   8   8  8
+        9   1    1      9   9   9  9
+        """,
+        priority=0,
```

Comment:
Please double check the priority.

Reply:
I think this case has requirement for this numa count, if use a minimal requirement, we may miss this check.

Reply:
This is P1. I was using it for easier debugging but will modify it to 1 in the next commit.

Reply:
@LiliDeng I didn't get you second comment. Can you please elaborate?